### DISCUSSION TITLE
OllamaGenerateRequest with JSON format

### DISCUSSION THREAD

**Discussion by seeseemelk** [amithkoujalgi/ollama4j#66] [Q&A]
I'm trying to generate a JSON response.
As part of another issue (https://github.com/ollama4j/ollama4j/issues/24), it was supposedly added using the "returnFormatJson" property on OllamaGenerateRequest.

However, I can't seem to find how to actually use it. As far as I can tell, while it has been added internally, it's not actually exposed using the publicly facing API's.

**Comment by amithkoujalgi** [comment]
Hi @seeseemelk. You're right. The parameter `returnFormatJson` was indeed added but it isn't being made use of in the APIs. Thanks for pointing it out.

It was designed to instruct the LLM to respond with JSON format (via the `format` param available in Ollama).

But the feature from Ollama seems to be buggy and you could observe that when you directly access Ollama REST APIs.

For example:

```shell
curl --silent http://localhost:11434/api/generate -d '{
  "model": "llama3.1",
  "prompt": "Why is the sky blue?",
  "format" : "json",
  "stream" : false
}' | jq -r .response
```

As a workaround, you could generate JSON responses through prompting.

In your prompt, mention that `you'd like the responses to be in JSON format with three backticks` and then you can capture the response string and parse out the JSON.
